Add new columns for supplement gating questions

quazi-h opened the pull request:
## Why
The gating question for whether daily supplements or infrequent supplements are not in the hles_dog table.
As it stands now, a data user would have to check 10 other variables just to determine if daily supplements or infrequent supplements were given.
[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1462)

## This PR
- Added two new columns to hles_dog_diet json schema: df_daily_supplements, df_infrequent_supplements
- Added gating logic to the mapDailySupplements and mapInfrequentSupplements functions within DietTransformations
- Updated unit tests

quazi-h committed:
Added two new columns to hles_dog_diet json schema: df_daily_supplements, df_infrequent_supplements
Added some gating logic to the mapDailySupplements and mapInfrequentSupplements functions within DietTransformations
Updated unit tests

quazi-h committed:
Merge branch 'master' into QH-DSPDC-1462-add-supplement-gates

ThePletch commented on hles/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/DietTransformations.scala:
```diff
@@ -156,77 +156,96 @@ object DietTransformations {
-        rawRecord.getOptional("df_s_other_text")
+  /** Map fields about supplements taken daily by a dog when available. */
+  def mapDailySupplements(rawRecord: RawRecord, dog: HlesDogDiet): HlesDogDiet =
+    rawRecord.getOptionalBoolean("df_supplement_daily").fold(dog) { dailySupps =>
```
we can probably turn this into a proper `if/else` statement instead of wrapping it in the `fold` by using `getRequired` instead of `getOptionalBoolean` here, since this field is required.

ThePletch reviewed (approved):
couple of suggestions about optional vs required (since the two yes/no questions here are both marked as 'required' in redcap), but otherwise looks good

quazi-h committed:
Merge branch 'master' into QH-DSPDC-1462-add-supplement-gates

quazi-h committed:
Merge branch 'master' into QH-DSPDC-1462-add-supplement-gates

quazi-h committed:
Extended RawRecord with a new getRequiredBoolean method.
Pulling the gating supplement questions with this new getRequiredBoolean to refactor the code to simplify branching logic.

quazi-h committed:
Updating unit tests.

Commit:
fix tests
